clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -226,21 +226,6 @@ record_filter_t::parallel_shard_memref(void *shard_data, const trace_entry_t &in
                 entry.addr |= OFFLINE_FILE_TYPE_BIMODAL_FILTERED_WARMUP;
             }
             break;
-        case TRACE_MARKER_TYPE_TIMESTAMP:
-            // No need to remember the previous unit's header anymore. We're in the
-            // next unit now.
-            // XXX: it may happen that we never output a unit header due to this
-            // optimization. We should ensure that we output it at least once. We
-            // skip handling this corner case for now.
-            per_shard->last_delayed_unit_header.clear();
-            ANNOTATE_FALLTHROUGH;
-        case TRACE_MARKER_TYPE_WINDOW_ID:
-        case TRACE_MARKER_TYPE_CPU_ID:
-            // Optimize space by outputting the unit header only if we are outputting
-            // something from that unit.
-            if (output)
-                per_shard->last_delayed_unit_header.push_back(entry);
```

Comment:
I think all uses of `last_delayed_unit_header` can be removed as it only exists to avoid duplicate unit headers.  Let's bring @abhinav92003 in on this as he wrote this code.

Reply:
Right. It can be removed now.

Reply:
I don't see the other references being removed?

Reply:
@huyangh-430 Please remove the other refs to last_delayed_unit_header; e.g., see L264 below which will always be false now.

Reply:
Removed code about last_delayed_unit_header, L264-268 in record_filter.cpp

Reply:
Please remove all instances: search for it.  It's still in the header.

Reply:
Removed another instance from record_filter.h.